src/pymedphys/dicom/_level3/dicom_collection.py

```diff
@@ -0,0 +1,162 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+
+from copy import deepcopy
+
+import numpy as np
+import pydicom
+
+from .._level1.create import dicom_dataset_from_dict
+from .._level2.anonymise import anonymise_dicom_dataset
+from .._level2.dose import (
+    extract_dicom_patient_xyz, convert_xyz_to_dicom_coords)
+
+from ...libutils import get_imports
+
+IMPORTS = get_imports(globals())
+
+# pylint: disable=W0201
+
+
+# Many thanks to the Medical Connections for offering free
+# valid UIDs (http://www.medicalconnections.co.uk/FreeUID.html)
+# Their service was used to obtain the following root UID for PyMedPhys:
+PYMEDPHYS_ROOT_UID = '1.2.826.0.1.3680043.10.188'
+
+
+class DicomBase:
+    def __init__(self, dataset, copy=True):
+        if copy:
+            dataset = deepcopy(dataset)
+
+        if dataset.is_little_endian is None:
+            dataset.is_little_endian = True
+
+        if dataset.is_implicit_VR is None:
+            dataset.is_implicit_VR = True
+
+        self.dataset = dataset
+
+    @classmethod
+    def from_file(cls, filepath):
+        dataset = pydicom.read_file(filepath, force=True)
+
+        return cls(dataset)
+
+    def to_file(self, filepath):
+        self.dataset.save_as(filepath)
+
+    @classmethod
+    def from_dict(cls, dictionary):
+        dataset = dicom_dataset_from_dict(dictionary)
+
+        return cls(dataset)
+
+    def __repr__(self):
+        return self.dataset.__repr__()
+
+    def __eq__(self, other):
+        return self.dataset.__eq__(other.dataset)
+
+    def anonymise(self, inplace=False):
+        to_copy = not inplace
+        anonymised = anonymise_dicom_dataset(self.dataset, copy=to_copy)
+
+        if not inplace:
+            return self.__class__(anonymised)
+
+
+class DicomDose(DicomBase):
+
+    def __init__(self, dataset, copy=True):
+        super().__init__(dataset, copy=copy)
+
+        self.mask = None
+
+    @property
+    def values(self) -> np.ndarray:
+        return self.dataset.pixel_array * self.dataset.DoseGridScaling
+
+    @property
+    def units(self):
+        return self.dataset.DoseUnits
+
+    @units.setter
+    def units(self, units_used):
```

Comment:
Not sure I like having this function. A change in units would require a change in values (from those at instantiation) and there's no check of that here...

Reply:
Yup, let's remove it.